### ISSUE TITLE
Getting interface {} is string, not *string all of a sudden

### ISSUE THREAD

**Issue by segersniels** [jszwedko/go-circleci#28]
```
panic: interface conversion: interface {} is string, not *string [recovered]
panic: interface conversion: interface {} is string, not *string
github.com/jszwedko/go-circleci/circleci.go:756
```

**Comment by segersniels** [comment]
Changing the type from *string to string seems to just fix this issue without any noticeable issues

```
MemoryLimit            string  `json:"memory-limit"`
if v, ok := f.raw["memory-limit"]; ok {
	if v != nil {
		f.MemoryLimit = v.(string)
	}
}
```

**Comment by segersniels** [comment]
Pull request: #29

**Comment by jszwedko** [comment]
Thanks for reporting this and submitting a fix @segersniels 

I can see how it will not correctly handle memory-limit or fleet if it is present and not null since the JSON decoder will parse it as a value rather than a pointer.

Your fix in #29 raises the broader question of whether it's useful to know if any of the flags are absent, or set to null, in the response, as changing the field from a `*string` to a `string` loses this information; missing or null values in the response will just show up as `""`. However, this is currently the case for all of the `bool` feature flags (missing or null values will show up as the default boolean value: `false`).

I opened #31 for an alternative approach which preserves the ability to determine if `fleet` or `memory-limit` are absent (or null) vs. empty. This also preserves backwards compatibility. Would that also satisfy your use case?

On a related note, could you provide a snippet of the actual JSON response with `memory-limit` set? It'd be useful to know what sorts of values to expect there as none of my builds have that feature flag field.

**Comment by segersniels** [comment]
#31 is probably a far better solution than directly changing the type, this solution fixes my issue so I closed #29.

As for the snippet I don't think I fully understand what you mean. My use case is fairly simple where I just list circle builds. This just suddenly stopped working one day, which probably means something changed on CircleCI's end?

**Comment by jszwedko** [comment]
Thanks @segersniels !

I'm mostly curious what value you get in the `memory-limit` field; e.g. is it something like `512MB`?

**Comment by segersniels** [comment]
When I log `v` in `UnmarshalJSON` I get the following:

```
if v, ok := f.raw["memory-limit"]; ok {
		util.Log(v)
		if v != nil {
			f.MemoryLimit = v.(*string)
		}
}
```
```
INFO[0001] (string) (len=2) "8G"
```

Underneath is a snippet of the feature flags associated with the project after the fix.

```
FeatureFlags: (circleci.FeatureFlags) {
  TrustyBeta: (bool) true,
  OSX: (bool) false,
  SetGithubStatus: (bool) true,
  BuildPRsOnly: (bool) false,
  ForksReceiveSecretVars: (bool) true,
  Fleet: (*string)(<nil>),
  BuildForkPRs: (bool) false,
  AutocancelBuilds: (bool) false,
  OSS: (bool) false,
  MemoryLimit: (*string)(<nil>),
  raw: (map[string]interface {}) (len=9) {
   (string) (len=3) "osx": (bool) false,
   (string) (len=17) "set-github-status": (bool) true,
   (string) (len=14) "build-fork-prs": (bool) false,
   (string) (len=11) "trusty-beta": (bool) true,
   (string) (len=14) "builds-service": (bool) false,
   (string) (len=14) "build-prs-only": (bool) false,
   (string) (len=29) "forks-receive-secret-env-vars": (bool) true,
   (string) (len=5) "fleet": (interface {}) <nil>,
   (string) (len=17) "autocancel-builds": (bool) false
  }
 },
```

**Comment by jszwedko** [comment]
Gotcha, thanks for the additional info @segersniels ! It's helpful to see examples since the API documentation lacks detail.

**Comment by jszwedko** [comment]
Will close this out since it should be fixed by #31 , but let me know if you see any issues.
